src/external-server/docker-amarna.ts

```diff
@@ -1,27 +1,24 @@
-import { Image } from "@nomiclabs/hardhat-docker";
-import { DockerServer } from "./docker-server";
+import { HardhatDocker, Image } from "@nomiclabs/hardhat-docker";
 import * as fs from "fs";
-import { spawnSync } from "child_process";
-import { stdout } from "process";
 
-export class AmarnaDocker extends DockerServer {
+export class AmarnaDocker {
     useShell = false;
+    container: string;
+    docker: HardhatDocker;
     /**
      * @param image the Docker image to be used for running the container
      * @param cairoPaths the paths specified in hardhat config cairoPaths
      */
-    constructor(image: Image, private path: string) {
-        super(image, "127.0.0.1", null, "", "amarna-docker");
+    constructor(private image: Image, private path: string) {
+        this.container = "amarna-container-" + Math.random();
     }
 
-    protected async getDockerArgs(): Promise<string[]> {
-        // To access the files on host machine from inside the container, proper mounting has to be done.
-        const volumes = ["-v", `${this.path}:/src`];
-        const dockerArgs = [...volumes];
+    protected getDockerArgs(): string[] {
+        let cmd = ["amarna", ".", "-o", "out.sarif"];
```

Comment:
I'd also use slicing to remove `"0."` from the name, as we do in `DockerServer`:
```js
Math.random().toString().slice(2);
```